XamarinCommunityToolkit/Extensions/ObservableCollectionExtension.shared.cs

```diff
@@ -0,0 +1,36 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+
+namespace Microsoft.Toolkit.Xamarin.Forms.Extensions
+{
+    /// <summary>
+    /// Extension methods for <see cref="ObservableCollection{T}"/>.
+    /// </summary>
+    public static class ObservableCollectionExtension
+    {
+        /// <summary>
+        /// Adds a range of items to a <paramref name="collection"/>.
+        /// </summary>
+        /// <param name="collection">Collection where the items will be added to.</param>
+        /// <param name="items">Items to add to the collection.</param>
+        public static void AddRange<T>(this ObservableCollection<T> collection, IEnumerable<T> items)
+        {
+            foreach (var item in items)
+            {
+                collection.Add(item);
```

Comment:
With this implementation each add would still generate a notification... I always tend to use a subclassed RangeObservableCollection that overrides the OnCollectionChanged to suppress the single add notifications.

Reply:
That is correct. You can just steal my ObservableRangeCollection. Steal all of Mvvmhelpers please!